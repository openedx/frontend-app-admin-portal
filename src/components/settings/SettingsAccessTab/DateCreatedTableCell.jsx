import { useMemo } from 'react';
import PropTypes from 'prop-types';

import { formatTimestamp } from '../../../utils';

const DateCreatedTableCell = ({ row }) => {
  const { created } = row.original;
  const formattedDateCreated = useMemo(() => formatTimestamp({ timestamp: created }), [created]);
  return formattedDateCreated;
};

DateCreatedTableCell.propTypes = {
  row: PropTypes.shape({
    original: PropTypes.shape({
      created: PropTypes.string,
    }),
  }).isRequired,
};

export default DateCreatedTableCell;
