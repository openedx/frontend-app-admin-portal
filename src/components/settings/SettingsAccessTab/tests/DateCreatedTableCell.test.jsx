import React from 'react';
import renderer from 'react-test-renderer';

import DateCreatedTableCell from '../DateCreatedTableCell';

describe('DateCreatedTableCell', () => {
  it('renders correctly', () => {
    const props = {
      row: {
        original: {
          created: '2022-01-10T12:00:00Z',
        },
      },
    };
    const tree = renderer
      .create(<DateCreatedTableCell {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
