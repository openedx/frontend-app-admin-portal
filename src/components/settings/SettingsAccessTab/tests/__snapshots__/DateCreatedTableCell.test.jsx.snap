// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateCreatedTableCell renders correctly 1`] = `"January 10, 2022"`;
